frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx

```diff
@@ -761,15 +784,56 @@ export const CRDCardRow = connect(crdCardRowStateToProps)((props: CRDCardRowProp
   );
 });
 
+const AlertToCreateCustomResource: React.SFC<AlertToCreateCustomResourceProps> = (props) => {
+  const { cr } = props;
+
+  const crName = cr?.[0]?.kind;
+  const crDefinition = props.obj.spec?.customresourcedefinitions?.owned?.find(
+    (api) => api.kind === crName,
+  );
+  const crModelName = crDefinition?.name;
+  const { group: crGroup } = groupVersionFor(cr?.[0]?.apiVersion);
+  const { findModel } = useModelFinder();
+  const model = findModel(crGroup, crModelName.split('.').shift());
+
+  // Check if the CR is already present - only checks for the specific name/ns in the initialization-resource
+  const [customResource, customResourceLoaded] = useK8sGet<K8sResourceKind>(
+    model,
+    cr?.[0]?.metadata.name,
+    cr?.[0]?.metadata.namespace,
+  );
+
+  if (!customResource && customResourceLoaded) {
+    return (
+      <Alert isInline className="co-alert" variant="warning" title={`${crName} Required`}>
+        <p>Create a {crName} instance to use this operator.</p>
+        <CreateCRDButton obj={props.obj} />
```

Comment:
Can we pass the parsed JSON and the model through here instead of parsing the annotation again inside of `CreateCRDButton`? There's a lot of code duplication around handling this annotation.

I'm not sure how `CreateCRDButton` works since it won't have the model.

Reply:
Agreed.  The current `CreateCRDButton` component does need the obj for the name so I cannot remove obj completely.  I would like to refactor this function but I would need to refactor the install page component as well.  I will open a new Jira ticket to do those changes after this in order to prevent scope creep.